_posts/2021-11-05-r-6.0.0.md

```diff
@@ -0,0 +1,235 @@
+    dest == "LAS"
+    ) %>%
+  select(dep_time, arr_time, carrier) %>%
+  left_join(
+    arrow_table(nycflights13::airlines)
+   ) %>%
+  collect()
+
+#> # A tibble: 12 × 4
+#>    dep_time arr_time carrier name
+#>       <int>    <int> <chr>   <chr>
+#>  1      637      853 B6      JetBlue Airways
+#>  2      648      912 AA      American Airlines Inc.
+#>  3      812     1029 DL      Delta Air Lines Inc.
+#>  4      945     1206 VX      Virgin America
+#>  5      955     1219 B6      JetBlue Airways
+#>  6     1018     1231 DL      Delta Air Lines Inc.
+#>  7     1120     1338 B6      JetBlue Airways
+#>  8     1451     1705 DL      Delta Air Lines Inc.
+#>  9     1656     1915 AA      American Airlines Inc.
+#> 10     1755     2001 DL      Delta Air Lines Inc.
+#> 11     1827     2049 B6      JetBlue Airways
+#> 12     1917     2126 DL      Delta Air Lines Inc.
+```
+
+In this example, we're working on an in-memory table, so you wouldn't need `arrow` to do this--but the same code would work on a larger-than-memory dataset backed by thousands of Parquet files.
+
+## Under the hood
+
+To support these features, we've made some internal changes to how queries are built up and--importantly--when they are evaluated. As a result, there are some changes in behavior compared to past versions of `arrow`.
+
+First, calls to `summarise()`, `head()`, and `tail()` no longer eagerly evaluate: this means you need to call either `compute()` (to evaluate it and produce an Arrow Table) or `collect()` (to evaluate and pull the Table into an R `data.frame`) to see the results.
+
+Second, the order of rows in a dataset query is no longer determinisitic due to the way the parallelization of work happens in the C++ library. This means that you can't assume that the results of a query will be in the same order as the rows of data in the files on disk. If you do need a stable sort order, call `arrange()` to specify ordering.
+
+While these changes are a break from past `arrow` behavior, they are consistent with many `dbplyr` backends and are needed to allow queries to scale beyond data-frame workflows that can fit into memory.
+
+# Integration with DuckDB
+
+The Arrow engine is not the only new way to query Arrow Datasets in this release. If you have the [duckdb](https://duckdb.org/) package installed, you can hand off an Arrow Dataset or query object to duckdb for further querying using the `to_duckdb()` function. This allows you to use duckdb's `dbplyr` methods, as well as its SQL interface, to aggregate data. DuckDB supports filter pushdown, so you can take advantage of Arrow Datasets and Arrow-based optimizations even within a DuckDB SQL query with a `where` clause. Filtering and column projection specified before the `to_duckdb()` call in a pipeline is evaluated in Arrow; this can be helpful in some circumstances like complicated dbplyr pipelines.  You can also hand off DuckDB data (or the result of a query) to arrow with the `to_arrow()` call.
```

Comment:
```suggestion
The Arrow engine is not the only new way to query Arrow Datasets in this release. If you have the [duckdb](https://cran.r-project.org/package=duckdb) package installed, you can hand off an Arrow Dataset or query object to [DuckDB](https://duckdb.org/) for further querying using the `to_duckdb()` function. This allows you to use duckdb's `dbplyr` methods, as well as its SQL interface, to aggregate data. DuckDB supports filter pushdown, so you can take advantage of Arrow Datasets and Arrow-based optimizations even within a DuckDB SQL query with a `where` clause. Filtering and column projection specified before the `to_duckdb()` call in a pipeline is evaluated in Arrow; this can be helpful in some circumstances like complicated dbplyr pipelines.  You can also hand off DuckDB data (or the result of a query) to arrow with the `to_arrow()` call.
```